ivy_tests/test_docstrings.py

```diff
@@ -71,7 +71,13 @@ def test_docstrings(backend):
         if k == "Array":
             for method_name in dir(v):
                 method = getattr(ivy.Array, method_name)
-                if method_name in skip_arr_cont or helpers.docstring_examples_run(
+                try:
+                    gradient_incompatible = method.computes_gradients and not backend.supports_gradients
+                except AttributeError:
```

Comment:
we shouldn't use `try` `except` for this, we can do something like `gradient_incompatible = not backend.supports_gradients and hasattr(method, "computes_gradients") and method.computes_gradients`

we always evaluate expressions in `and`, `or` etc. from left to right, and this stops as soon as an evaluation leads to `False`, meaning that `method.computes_gradients` will not be evaluated if `hasattr(method, "computes_gradients")` returns `False`.

But actually, we should implement a simple dedicated helper function for this called `gradient_incompatible_function`, in a similar vein to [function_unsupported_dtypes](https://github.com/unifyai/ivy/blob/3dda99b1ed295e0ac67de7c4170dd041c23b78bc/ivy/functional/ivy/data_type.py#L1134) etc.

This helper function implementation can be as simple as:

```
return not ivy.supports_gradients and hasattr(fn_in, "computes_gradients") and fn_in.computes_gradients
```